### PR TITLE
add lowercase versions of the constants

### PR DESCRIPTION
**Pull request by devoncarew**
- add lowercase versions of the logging constants
- fix https://github.com/dart-lang/logging/issues/43

@pq @kevmoo

### CODE REVIEW

**Comment by kevmoo** [comment]
CC @natebosch @jakemac53 @grouma 

Thoughts on this?

I hate adding the other set of constants without deprecating the old ones.

I also cringe at all of the deprecation warnings it'll cause...hrm...

**Comment by pq** [comment]
My sense is we should bite the bullet and deprecate.  Updating references will be a bummer but a one-time thing.  I'm happy to pitch in some of the community packages too.

File: lib/logging.dart [hunk 1]
```diff
@@ -271,48 +271,61 @@ class Level implements Comparable<Level> {
   const Level(this.name, this.value);
 
   /// Special key to turn on logging for all levels ([value] = 0).
-  static const Level ALL = const Level('ALL', 0);
+  static const Level all = const Level('ALL', 0);
+  static const Level ALL = all;
```
Comment (pq) [inline lib/logging.dart]: I'd suggest we go ahead and `@deprecate` all these.

File: lib/logging.dart [hunk 2]
```diff
@@ -271,48 +271,61 @@ class Level implements Comparable<Level> {
...
+    fine,
+    config,
+    info,
+    warning,
+    severe,
+    shout,
+    off,
   ];
 
+  static const List<Level> LEVELS = levels;
```
Comment (pq) [inline lib/logging.dart]: And deprecate this too. 👍

**Comment by jakemac53** [comment]
My feeling is this provides to little value to justify the churn it causes. This is forcing real work on lots of people for what is the equivalent of a style nit.

**Comment by devoncarew** [comment]
> I hate adding the other set of constants without deprecating the old ones.

The all caps constants look old; definitely not a P1 to rename them though. I didn't deprecate the old ones just because of the cost downstream.

> I also cringe at all of the deprecation warnings it'll cause...hrm...

Deprecating them would cause a lot of downstream work - likely more work than we'd actually want to do.

**Comment by kevmoo** [comment]
> arguably negative value because now there are 2 ways to do exactly the same thing

AGREED!

**Comment by devoncarew** [comment]
I'll close this, but we may want to consider a pre-deprecation stage in cases like this. I'd would be useful to be able to switch clients to the new thing, w/o deprecation messages existing yet on the old one.

**Pull request closed without merging**